src/allotropy/parsers/mabtech_apex/mabtech_apex_structure.py

```diff
@@ -41,29 +43,65 @@ def create_metadata(plate_info: SeriesData, file_name: str) -> Metadata:
 def _create_measurement(plate_data: SeriesData) -> Measurement:
     location_id = plate_data[str, "Well"]
     well_plate = plate_data.get(str, "Plate")
-
-    return Measurement(
+    led_filter = plate_data.get(str, "LED Filter")
+    led_filter_without_total = led_filter.split(" ")[0] if led_filter else None
+    exposure_duration_setting_key = (
+        f"{led_filter_without_total} Exposure"
+        if led_filter_without_total
+        else "Exposure"
+    )
+    illumination_setting_key = (
+        f"{led_filter_without_total} Preset Intensity"
+        if led_filter_without_total
+        else "Preset Intensity"
+    )
+    measurement = Measurement(
         type_=MeasurementType.OPTICAL_IMAGING,
         identifier=random_uuid_str(),
         location_identifier=location_id,
         well_plate_identifier=well_plate,
         sample_identifier=f"{well_plate}_{location_id}",
         detection_type=constants.DETECTION_TYPE,
         device_type=constants.DEVICE_TYPE,
-        exposure_duration_setting=plate_data.get(float, "Exposure"),
-        illumination_setting=plate_data.get(float, "Preset Intensity"),
+        led_filter=led_filter,
+        exposure_duration_setting=plate_data.get(float, exposure_duration_setting_key),
+        illumination_setting=plate_data.get(float, illumination_setting_key),
         processed_data=ProcessedData(
             identifier=random_uuid_str(),
             features=[
-                ImageFeature(
-                    identifier=random_uuid_str(),
-                    feature=feature,
-                    result=plate_data.get(float, feature, NaN),
-                )
+                value
                 for feature in constants.IMAGE_FEATURES
+                if (
+                    value := _build_feature(
+                        feature, plate_data, led_filter_without_total
+                    )
+                )
             ],
         ),
     )
+    if not (measurement.processed_data and measurement.processed_data.features):
+        logging.warning(f"no image features identified for {well_plate}")
+    return measurement
+
+
+def _build_feature(
+    feature: str, plate_data: SeriesData, led_filter: str | None
+) -> ImageFeature | None:
+    if led_filter:
+        led_number = re.search(r"\d+", led_filter)
+        if not led_number:
+            error_msg = f"Unable to interpret LED number from {led_filter}"
+            raise AllotropeParsingError(error_msg)
+        feature = feature.format(filter=led_filter, led_number=led_number.group())
+    else:
+        feature = feature.format(filter="", led_number="").replace("  ", " ")
+    if not plate_data.get(float, feature):
```

Comment:
Need to check if value `is not None`, see test data where you drop some values that are 0.

Also, you can fetch the value here and use it below:

```
if (value := plate_data.get(float, feature)) is None:
    return None
```